### PR TITLE
create new Prow cron job to update the vulnerability dashboard

### PR DESCRIPTION
**Pull request by yodahekinsew**
Created a new Prow cron job ("k8s-vuln-dashboard-cron") to run once every 24 hours to update the vulnerability dashboard. The vulnerability dashboard is described in more detail [here](https://github.com/kubernetes-sigs/k8s-container-image-promoter/issues/253). 

This Prow job will update the contents of the vulnerability dashboard by reading in vulnerability occurrences from the Container Analysis Service and writing to a json stored in GCS.

### CODE REVIEW

File: config/jobs/kubernetes/test-infra/test-infra-trusted.yaml [hunk 1]
```diff
@@ -1083,3 +1083,26 @@ periodics:
...
+    serviceAccountName: k8s-infra-gcr-vuln-dashboard
+    containers:
+    - image: us.gcr.io/k8s-artifacts-prod/artifact-promoter/cip:20200807-v2.3.1-258-gc35b3a0
+      command:
+      - dashboard
+      args:
+        - -dashboard-file-path=/home/prow/go/src/github.com/kubernetes-sigs/k8s-container-image-promoter/dashboard/
+        - -vuln-target-project=k8s-artifacts-prod
+        - -dashboard-bucket=gs://k8s-artifacts-prod-vuln-dashboard
+  annotations:
```
Comment (yodahekinsew) [inline config/jobs/kubernetes/test-infra/test-infra-trusted.yaml:1105]: Are these the correct annotations for this job? Not sure what test grid annotation to assign this job; I just copied values over from another periodic Prow job above this one.

Comment (yodahekinsew) [reply config/jobs/kubernetes/test-infra/test-infra-trusted.yaml:1105]: @listx @spiffxp

File: config/jobs/kubernetes/test-infra/test-infra-trusted.yaml [hunk 2]
```diff
@@ -1083,3 +1083,26 @@ periodics:
...
+    testgrid-dashboards: sig-release-releng-blocking, wg-k8s-infra-k8sio
+    testgrid-alert-email: k8s-infra-alerts@kubernetes.io, release-managers@kubernetes.io
```
Comment (spiffxp) [inline config/jobs/kubernetes/test-infra/test-infra-trusted.yaml:1107]: I'd like the k8s-infra stuff to be included (and we should soon work to have this running over there). 

I'm guessing this shouldn't block releng, but they'll want to know

```suggestion
    testgrid-dashboards: sig-release-releng-informing, wg-k8s-infra-k8sio
    testgrid-alert-email: k8s-infra-alerts@kubernetes.io, release-managers@kubernetes.io
```

**Comment by listx** [comment]
FWIW I vote to get this merged as-is as Yodahe is currently blocked on this PR.

**Comment by listx** [comment]
Ping?

File: config/jobs/kubernetes/test-infra/test-infra-trusted.yaml [hunk 3]
```diff
@@ -1083,3 +1083,26 @@ periodics:
     testgrid-dashboards: sig-release-releng-blocking
     testgrid-alert-email: k8s-infra-alerts@kubernetes.io, release-managers@kubernetes.io
     testgrid-num-failures-to-alert: '1'
+- interval: 24h
+  cluster: test-infra-trusted
+  max_concurrency: 1
+  name: k8sio-vuln-dashboard-cron
```
Comment (justaugustus) [inline config/jobs/kubernetes/test-infra/test-infra-trusted.yaml:1089]:
```suggestion
  name: ci-k8sio-vuln-dashboard-update
```

For periodic jobs, we usually prepend `ci-`.
Add `-update` as a suffix as well, so it's a little clearer what this job is doing at-a-glance.

**Pull request merged**